### PR TITLE
Added popup UI and UI functionality

### PR DESCRIPTION
**Pull request by PrvCaboose**
Tower and Spell buttons work and activate properly, so far only the first spell does something, but it doesn't affect the enemies yet, just collides with them. All Main UI buttons have a hover mechanic where an info panel shows up, and towers have a info panel show up on click, and disappear when they click off. Tower info panels remain within the bounds of the screen.

### CODE REVIEW

File: Defend The Divine/Assets/Scripts/Spells/FreezeSpell.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    private Vector3 direction;
+    private float speed = 5f;
+
+    public void SetTarget(Vector3 targetPosition)
+    {
+        direction = (targetPosition - transform.position).normalized;
+    }
+
+    public void SetStats(float damage, float speed)
```
Comment (AJKaz) [inline Defend The Divine/Assets/Scripts/Spells/FreezeSpell.cs:18]: don't need damage

File: Defend The Divine/Assets/Scripts/Towers/TowerPlacement.cs [hunk 2]
```diff
@@ -53,6 +58,19 @@ void Update()
...
+            switch (currentTowerType)
+            {
+                case TowerType.tower1:
+                    towerPrefab = towerType1Prefab;
+                    break;
+                case TowerType.tower2:
+                    towerPrefab = towerType2Prefab;
+                    break;
+                default:
+                    towerType2Prefab = null;
```
Comment (AJKaz) [inline Defend The Divine/Assets/Scripts/Towers/TowerPlacement.cs:70]: why do you set the prefab to null?

**Review by AJKaz — APPROVED** [review]
nice work, consider changing those few things tho

**Review by ArgentumRhodon — CHANGES_REQUESTED** [review]
I would address AJ's concerns before merging. I didn't find anything that he didn't already comment on.

**Pull request merged**